ruby/ql/lib/codeql/ruby/regexp/RegExpTreeView.qll

```diff
@@ -440,6 +456,24 @@ class RegExpSequence extends RegExpTerm, TRegExpSequence {
   }
 
   override string getAPrimaryQlClass() { result = "RegExpSequence" }
+
+  override predicate matchesEmptyString() {
+    forall(RegExpTerm child | child = this.getAChild() | child.matchesEmptyString())
+  }
+
+  // Why can't we use concat(...) with language[monotonicAggregates] here instead?
```

Comment:
This predicate looks like it could have very many results. I couldn't find an equivalent in the JS library, perhaps for a good reason.